downstairs/src/region.rs

```diff
@@ -440,28 +444,21 @@ impl Region {
             self.dirty_extents.insert(eid);
         }
 
-        *mg = ExtentState::Opened(Arc::new(new_extent));
+        *mg = ExtentState::Opened(new_extent);
 
         Ok(())
     }
 
     pub async fn close_extent(
-        &self,
+        &mut self,
         eid: usize,
     ) -> Result<(u64, u64, bool), CrucibleError> {
-        let mut extent_state = self.extents[eid].lock().await;
+        let extent_state = &mut self.extents[eid];
 
-        let open_extent =
-            std::mem::replace(&mut *extent_state, ExtentState::Closed);
+        let open_extent = std::mem::replace(extent_state, ExtentState::Closed);
 
         match open_extent {
-            ExtentState::Opened(extent) => {
-                // extent here is Arc<Extent>, and closing only makes sense if
-                // the reference count is 1.
-                let inner = Arc::into_inner(extent);
-                assert!(inner.is_some());
-                inner.unwrap().close().await
-            }
+            ExtentState::Opened(extent) => extent.close().await,
```

Comment:
I believe that in the existing code, this `Arc` was not used when you have multiple connections:

- No matter how many connections you have, there's only one `Downstairs` instance (shared with an `Arc<Mutex<Downstairs>>`, which is still present after this PR)
- This `Downstairs` owns a single `Region`
- This `Region` owns the individual `Extent`'s

I do see one way for the `Arc` to have multiple owners in the previous code: if someone used `get_opened_extent`, which returns a clone of the `Arc<Extent>`, and then held on to that value, then we'd panic here.

This PR makes that condition impossible, because `get_opened_extent` now returns a reference, so you can't hold on to it.

> Because this is now `mut`, we can't have multiple callers?

Yup, that's correct.